onnx2kerastl/reshape_layers.py

```diff
@@ -486,17 +487,26 @@ def convert_resize(node, params, layers, lambda_func, node_name, keras_name):
     else:
         raise Exception("unsupported resize method")
 
-    to_channel_last = keras.layers.Permute((2, 3, 1))(input_tensor)
+    axes = params.get('axes', list(range(input_tensor.shape.rank)))  # Default to resizing all axes
+
+    # Validate axes
+    rank = input_tensor.shape.rank
+    axes = [a if a >= 0 else a + rank for a in axes]  # Convert negative axes to positive
+    if any(a < 0 or a >= rank for a in axes):  # check that all axes values are within input rank
+        raise ValueError("Invalid axes value")
+
+    to_channel_last = keras.layers.Permute((2, 3, 1))(input_tensor)  # (B, W, H, C)
+    shape = tf.cast(tf.shape(to_channel_last), tf.int32)
+    tf_resize_shapes = [shape[i] for i in range(1, 3)]  # (W, H)
+
     if len(scales) > 0:
-        if scales[0] != 1 or scales[1] != 1:
-            raise Exception("Resize of channels or batch dim not suppported")
-        shape = tf.cast(tf.shape(to_channel_last), tf.float32)
-        tf_resize_shapes = [tf.cast(scales[2] * shape[1], tf.int32),
-                            tf.cast(scales[3] * shape[2], tf.int32)]
+        for i, axis in enumerate(axes):
+            if scales[i] != 1:
+                tf_resize_shapes[axis - 2] = tf.cast(scales[i] * shape[axis - 1], tf.int32)
```

Comment:
Might be good to change this to :
`tf_resize_shapes[axis - 2] = tf.cast(scales[i] * tf_resize_shapes[axis - 2], tf.int32)`
because it took me some time to understand index shift